diff --git a/netcat.c b/netcat.c
index b4e3ca6..f0f6cec 100644
--- a/netcat.c
+++ b/netcat.c
@@ -720,9 +720,7 @@ readwrite(int nfd)
 			if ((n = read(nfd, buf, plen)) < 0)
 				return;
 			else if (n == 0) {
-				shutdown(nfd, SHUT_RD);
-				pfd[0].fd = -1;
-				pfd[0].events = 0;
+				goto shutdown_rd;
 			} else {
 				if (tflag)
 					atelnet(nfd, buf, n);
@@ -730,17 +728,29 @@ readwrite(int nfd)
 					return;
 			}
 		}
+		else if (pfd[0].revents & POLLHUP) {
+		shutdown_rd:
+			shutdown(nfd, SHUT_RD);
+			pfd[0].fd = -1;
+			pfd[0].events = 0;
+		}
 
-		if (!dflag && pfd[1].revents & POLLIN) {
-			if ((n = read(wfd, buf, plen)) < 0)
-				return;
-			else if (n == 0) {
+		if (!dflag) {
+			if (pfd[1].revents & POLLIN) {
+				if ((n = read(wfd, buf, plen)) < 0)
+					return;
+				else if (n == 0) {
+					goto shutdown_wr;
+				} else {
+					if (atomicio(vwrite, nfd, buf, n) != n)
+						return;
+				}
+			}
+			else if (pfd[1].revents & POLLHUP) {
+			shutdown_wr:
 				shutdown(nfd, SHUT_WR);
 				pfd[1].fd = -1;
 				pfd[1].events = 0;
-			} else {
-				if (atomicio(vwrite, nfd, buf, n) != n)
-					return;
 			}
 		}
 	}
